File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesEpisodeGroups.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Episode groups for a TMDB TV series.
 *
 * @param id      TMDB TV series id
 * @param results episode groups
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesEpisodeGroups(
        int id,
        List<TvSeriesEpisodeGroup> results
) implements TmdbModel {

    public TvSeriesEpisodeGroups {
        results = TmdbCollections.list(results);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesEpisodeGroups.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Episode groups for a TMDB TV series.
 *
 * @property id TMDB TV series id
 * @property results episode groups
 */
@JvmRecord
data class TvSeriesEpisodeGroups(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("results")
    val results: List<TvSeriesEpisodeGroup> = emptyList(),
) : TmdbModel
